File: Example/FrameWorkDemo/ZallData/ZAViewsViewController.m
//
//  ZAViewsViewController.m
//  ZallData
//
//  Created by guo on 2022/1/26.
//  Copyright © 2022 Zall Data Co., Ltd. All rights reserved.
//

#import "ZAViewsViewController.h"

@interface ZAViewsViewController ()
@property (weak, nonatomic) IBOutlet UILabel *myLabel;
@property (weak, nonatomic) IBOutlet UIButton *myButton;
@property (weak, nonatomic) IBOutlet UIButton *myButtonItem;
@property (weak, nonatomic) IBOutlet UISegmentedControl *mySecond;
@property (weak, nonatomic) IBOutlet UITextField *myTextField;
@property (weak, nonatomic) IBOutlet UISlider *mySlider;
@property (weak, nonatomic) IBOutlet UISwitch *mySwitch;
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *myActivityIndicator;
@property (weak, nonatomic) IBOutlet UIProgressView *myProgressView;
@property (weak, nonatomic) IBOutlet UIPageControl *myPageControl;
@property (weak, nonatomic) IBOutlet UIStepper *myStepper;
@property (weak, nonatomic) IBOutlet UIImageView *myImageView;
@property (weak, nonatomic) IBOutlet UITextView *myTextView;
@property (weak, nonatomic) IBOutlet UIDatePicker *myDataPicker;

@end

@implementation ZAViewsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    UITapGestureRecognizer *imageViewTapGestureRecognizer = [[UITapGestureRecognizer alloc]initWithTarget:self action:@selector(imageViewTouchUpInside:)];
    self.myImageView.userInteractionEnabled = YES;
    [self.myImageView addGestureRecognizer:imageViewTapGestureRecognizer];
}

#pragma mark - Action
-(void)imageViewTouchUpInside:(id)sender{
    
}
- (IBAction)mybuttonAction:(id)sender {
    
}
- (IBAction)myButtonItemAction:(id)sender {
    
}
- (IBAction)mySecondAction:(id)sender {
    
}
- (IBAction)myTextFieldAction:(id)sender {
    
}

- (IBAction)mySliderAction:(UISlider *)sender {
    self.myProgressView.progress = sender.value;
}
- (IBAction)mySwitchAction:(UISwitch *)sender {
    if (sender.isOn) {
        [self.myActivityIndicator startAnimating];
    }else{
        [self.myActivityIndicator stopAnimating];
    }
    
}
- (IBAction)myPageAction:(id)sender {
    
}
- (IBAction)myStepper:(id)sender {
    
}

- (IBAction)myDataPicker:(id)sender {
    
}
 


-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    [self.view endEditing:YES];
}


/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Example/ZallDataSwift/ZAViewsViewController.swift
//
//  ZAViewsViewController.swift
//  ZallDataSwift
//
//  Created by Mac on 2022/1/26.
//  Copyright © 2022 Zall Data Co., Ltd. All rights reserved.
//

import UIKit

class ZAViewsViewController: UIViewController {
     
    @IBOutlet weak var myLabel:UILabel?
    @IBOutlet weak var myButton:UIButton?
    @IBOutlet weak var myButtonItem:UIButton?
    @IBOutlet weak var mySecond:UISegmentedControl?
    @IBOutlet weak var myTextField:UITextField?
    @IBOutlet weak var mySlider:UISlider?
    @IBOutlet weak var mySwitch:UISwitch?
    @IBOutlet weak var myActivityIndicator:UIActivityIndicatorView?
    @IBOutlet weak var myProgressView:UIProgressView?
    @IBOutlet weak var myPageControl:UIPageControl?
    @IBOutlet weak var myStepper:UIStepper?
    @IBOutlet weak var myImageView:UIImageView?
    @IBOutlet weak var myTextView:UITextView?
    @IBOutlet weak var myDataPicker:UIDatePicker?
   
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    
    @IBAction func mybuttonAction(_ sender: Any){
        
    }
    @IBAction func myButtonItemAction(_ sender: Any){
        
    }
    @IBAction func mySecondAction(_ sender: Any){
        
    }
    @IBAction func myTextFieldAction(_ sender: Any){
        
    }
    @IBAction func mySliderAction(_ sender: UISlider){
        self.myProgressView?.progress = sender.value;
    }
    @IBAction func mySwitchAction(_ sender: UISwitch){
        if (sender.isOn) {
            self.myActivityIndicator?.startAnimating()
        }else{
            self.myActivityIndicator?.stopAnimating()
        }
    }
    @IBAction func myPageAction(_ sender: Any){
        
    }
    @IBAction func myStepper(_ sender: Any){
        
    }
    @IBAction func myDataPicker(_ sender: Any){
        
    }
  

}
